src/main/java/net/neoforged/neoform/runtime/artifacts/ArtifactManager.java

```diff
@@ -59,15 +65,29 @@ public Artifact get(MinecraftLibrary library) throws IOException {
             throw new IllegalArgumentException("Cannot download a library that has no artifact defined: " + library);
         }
 
-        // TODO: if we identify where the Minecraft installation is, we could try to copy the library from there
-
         var artifactCoordinate = MavenCoordinate.parse(library.artifactId());
         var externalArtifact = getFromExternalManifest(artifactCoordinate);
         if (externalArtifact != null) {
             return externalArtifact;
         }
 
-        var finalLocation = artifactsCache.resolve(artifactCoordinate.toRelativeRepositoryPath());
+        var relativePath = artifactCoordinate.toRelativeRepositoryPath();
+
+        // Try reusing it from a local Minecraft installation, which ultimately is structured like a Maven repo
+        var localMinecraftLibraries = new ArrayList<>(launcherInstallations.getInstallationRoots());
+        for (var localRepo : localMinecraftLibraries) {
+            var localPath = localRepo.resolve("libraries").resolve(relativePath);
+            try {
+                // Ensure the file matches before using it
+                var fileHash = HashingUtil.hashFile(localPath, artifact.checksumAlgorithm());
+                if (Objects.equals(fileHash, artifact.checksum())) {
+                    return getArtifactFromPath(localPath);
+                }
+            } catch (IOException ignored) {
```

Comment:
Note that the reason this doesn't log is because it doesn't check if the file exists.